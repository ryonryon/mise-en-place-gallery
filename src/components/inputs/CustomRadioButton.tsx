import styled from "@emotion/styled";
import * as React from "react";

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {}

const CustomRadioButton = ({ ...rest }: Props) => {
  const StyledLabel = styled.label`
    color: #858585;
    position: relative;
    padding: 0 0 0 24px;

    &::after,
    &::before {
      position: absolute;
      content: "";
      display: block;
      top: 50%;
    }

    &::after {
      left: 15px;
      margin-top: -10px;
      width: 16px;
      height: 16px;
      border: 2px solid rgba(225, 225, 225, 0.3);
      border-radius: 50%;
    }

    &::before {
      left: 20px;
      margin-top: -5px;
      width: 10px;
      height: 10px;
      background: rgba(225, 225, 225, 0);
      border-radius: 50%;
    }

    &:hover::after {
      border: 2px solid rgba(225, 225, 225, 0.9);
      transition: all 0.5s ease-out;
    }

    &:hover::before {
      background: rgba(225, 225, 225, 0.9);
      transition: all 0.5s ease-out;
    }

    & > input[type="radio"] {
      display: none;
    }
  `;
  return (
    <StyledLabel>
      <input type={"radio"} {...rest} />
    </StyledLabel>
  );
};

export default CustomRadioButton;
