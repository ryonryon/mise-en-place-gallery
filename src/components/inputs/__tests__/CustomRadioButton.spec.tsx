import * as React from "react";
import * as renderer from "react-test-renderer";

import CustomRadioButton from "../CustomRadioButton";

describe("CustomRadioButton test", () => {
  it("tests CustomRadioButton renders correctly ", () => {
    const customRadioButton = renderer.create(<CustomRadioButton />).toJSON();
    expect(customRadioButton).toMatchSnapshot();
  });
});
