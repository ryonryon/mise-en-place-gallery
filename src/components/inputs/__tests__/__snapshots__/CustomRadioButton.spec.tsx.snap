// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadioButton test tests CustomRadioButton renders correctly  1`] = `
<label
  className="css-q3hjqi"
>
  <input
    type="radio"
  />
</label>
`;
